wrappers/csharp/src/Frame.cs

```
@@ -0,0 +1,221 @@
+            Dispose(true);
+            // TODO: uncomment the following line if the finalizer is overridden above.
+            GC.SuppressFinalize(this);
+        }
+        #endregion
+
+        public void Release()
+        {
+            if(m_instance.Handle != IntPtr.Zero)
+                NativeMethods.rs2_release_frame(m_instance.Handle);
+            m_instance = new HandleRef(this, IntPtr.Zero);
+        }
+
+        public Frame Clone()
+        {
+            object error;
+            NativeMethods.rs2_frame_add_ref(m_instance.Handle, out error);
+            return new Frame(m_instance.Handle);
+        }
+
+        public int Width
+        {
+            get
+            {
+                object error;
+                var w = NativeMethods.rs2_get_frame_width(m_instance.Handle, out error);
+                return w;
+            }
+        }
+
+        public int Height
+        {
+            get
+            {
+                object error;
+                var w = NativeMethods.rs2_get_frame_height(m_instance.Handle, out error);
+                return w;
+            }
+        }
+
+        public IntPtr Data
+        {
+            get
+            {
+                object error;
+                return NativeMethods.rs2_get_frame_data(m_instance.Handle, out error);
+            }
+        }
+
+        public StreamProfile Profile
+        {
+            get
+            {
+                object error;
+                return new StreamProfile(NativeMethods.rs2_get_frame_stream_profile(m_instance.Handle, out error));
+            }
+        }
+
+        public ulong Number
+        {
+            get
+            {
+                object error;
+                var frameNumber = NativeMethods.rs2_get_frame_number(m_instance.Handle, out error);
+                return frameNumber;
+            }
+        }
+
+        public double Timestamp
+        {
+            get
+            {
+                object error;
```

Q:are you ignoring the error object on purpose? What do we have to pass it and not remove it from the nativeMethods signature?

---

The custom marshaller is taking care of this object. If not null is returned, the marshaller will throw